### ISSUE TITLE
elections and voting for f26 are done, but my picture is missing completely

### ISSUE THREAD

**Issue by decathorpe** [fedora-infra/nuancier#113]
I don't know how / why it happened (it might be my fault, though I don't see how), but my submission for the F26 wallpaper election didn't show up in the voting and doesn't show up in the results, and is still marked as "new" in the "Your Submissions" tab. I didn't get the badge for submitting a picture, either, if that helps.

Picture title is: Frozen
FAS account name is decathorpe, too

**Comment by pypingou** [comment]
I've heard there has been a problem with rejecting some of the wallpaper at the end, so if your submission didn't make it, it was likely because either: it was submitted too late and couldn't be reviewed in time or it was rejected but a bug prevented it from being marked as such.

Maybe @gnokii would know which of these two it was for you :)

**Comment by decathorpe** [comment]
I submitted my picture March 22 or 23, I don't remember exactly - so that was kind of late.

Nonetheless, if submissions are *open* until March 23, I would expect that all timely submissions are taken into consideration.

**Comment by pypingou** [comment]
Did you see https://fedoramagazine.org/give-fedora-26-a-nice-look-contribute-your-wallpaper/ more precisely the ``Dates and deadlines`` section?
It explains that while your expectations are understandable, they are not correct at the moment.

**Comment by decathorpe** [comment]
I think I saw this blog post when it was published, but I assumed that the nuancier web app displayed correct information about the submission details and deadline. I understand that a legal review of the submissions is necessary and that this takes some time.

What I don't understand is why submissions on the web app were even open until March 23, if the "official" deadline was March 22.

**Comment by pypingou** [comment]
Could it be related to timezone? Nuancier uses UTC times, so depending on your timezone you could maybe submit something on day +1 while nuancier would turn off submission on day 0.

**Comment by decathorpe** [comment]
I'm in UTC+1 (UTC+2 with DST), so that's not the issue.

**Comment by pypingou** [comment]
So March 22nd was maybe a little late and maybe your submission didn't get reviewed, sorry.

https://apps.fedoraproject.org/nuancier/elections/

On March 23rd though I'm really doubting that submitting a candidate was possible since the election was open on that day.

**Comment by decathorpe** [comment]
Yes, and I understand that.

Still, according to your comments, this line of code is wrong (because it's the day the elections start, and not the day submissions close):

https://github.com/fedora-infra/nuancier/blob/master/nuancier/templates/contribute_index.html#L173

**Comment by pypingou** [comment]
> Still, according to your comments, this line of code is wrong (because it's the day the elections start, and not the day submissions close):

That's the issue, these are the same dates, as soon as the submission is closed, the election starts, which is why submission made close to the deadline may not be reviewed in time.

This is something we really ought to fix for the next election to ensure we have a submission start/end and an election start/end and that there can be a time between submission and the election.

**Comment by decathorpe** [comment]
Sounds like a good idea :+1:

**Comment by gnokii** [comment]
the answer is very simple, 22nd for a submission

https://fedoramagazine.org/give-fedora-26-a-nice-look-contribute-your-wallpaper/

states: Important note, submissions during the last hours can in certain circumstances not get into the election, if there is no time to do the legal research. 

There is already since September an issue open, to divide the end of the submission and the start of the election time. I had already for F25 a hard time to moderate all last minute submissions, please understand that it is not just hit a button to moderate a submission. Each submission needs to be checked if it is correct licensed, that is tricky and takes time.
Unfortunately more and more submissions come later now, even when we will have this division it might happen. The phase might be 3 days between the end of the submission and the start of the election, we have to cut therefore from the submission phase.

**Comment by decathorpe** [comment]
As I said, I understand the problem. I'm just a bit sad that my picture didn't make it (possibly) into F26, I'll try to submit my picture early next time ;)

> There is already since September an issue open, to divide the end of the submission and the start of the
> election time.

This seems like a good idea, especially since - for the F26 election - a "wrong" end of submissions date was displayed on nuancier's /contribute page (which confused me).

I'm going to close this issue, since it was resolved.

PS: I know my python, and I've already worked on small web applications - so if you want / need some help, I'd be happy to fix issues / implement small features :)
